### DISCUSSION TITLE
Detailed stack trace: Error: cannot find module quasar

### DISCUSSION THREAD

**Discussion by b0ot** [quasarframework/quasar#10571] [General - Components / Directives / etc] [answered]
I'm trying to update my cloud functions which use quasar. The functions work great when I'm testing them locally with the firebase emulators but they fail to deploy. The logs haven't been super helpful but I have found this line:
**_Detailed stack trace: Error: Cannot find module 'quasar'_**

**index.js**

> const functions = require('firebase-functions');
> const admin = require('firebase-admin');
> const nodemailer = require('nodemailer');
> const quasar = require('quasar')
> const {date , uid} = quasar
> const cors = require('cors')({
>   origin: true
> });
> 
> const sgMail = require('@sendgrid/mail')
> 
> 
> const db = admin.initializeApp().firestore()

package.json

>   "engines": {
>     "node": "12"
>   },
>   "main": "index.js",
>   "dependencies": {
>     "@quasar/cli": "^1.1.2",
>     "@sendgrid/mail": "^7.4.6",
>     "cors": "^2.8.5",
>     "firebase-admin": "^9.11.1",
>     "firebase-functions": "^3.15.5",
>     "nodemailer": "^6.6.3",
>     "nodeman": "^1.1.2"
>   },
>   "devDependencies": {
>     "firebase-functions-test": "^0.2.3"
>   },

I have done additional testing and it seems like the issue is related to quasar. Not sure what is different now but still causing issues.

**Comment by ontwikkelfabriek** [comment, marked as answer]
@b0ot 
> I'm trying to update my cloud functions which use quasar

Cloud functions using Quasar? That sounds to me like something is very wrong, Quasar being a **frontend** framework...

**Comment by hawkeye64** [comment]
@b0ot Doesn't look to me like you have quasar installed.

You actually have `"@quasar/cli": "^1.1.2",` installed. You need `quasar` and `@quasar/app` and remove `@quasar/cli`.

**Reply by b0ot** [reply]
Thanks for the answers.

The simplest solution for me was to remove Quasar as a dependency. I replaced date with npm package dateformat and uid with npm package nanoid. The general feedback I got here was that:

1.) Quasar is the wrong tool for the job and should be used for the frontend only.
2.) Even if you could use quasar you would be importing way more than needed and is best to use some lightweight zero dependency packages for these use cases.

I appreciate the responses @hawkeye64 & @ontwikkelfabriek ontwikkelfabriek
